samples/browseTable.js

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+function main(
+  datasetId = 'my_dataset', // Existing dataset
+  tableId = 'my_table' // Existing table
+) {
+  // [START bigquery_browse_table]
+  // Import the Google Cloud client library using default credentials
+  const {BigQuery} = require('@google-cloud/bigquery');
+  const bigquery = new BigQuery();
+
+  async function browseTable() {
+    // Retrieve a table's rows using manual pagination.
+
+    /**
+     * TODO(developer): Uncomment the following lines before running the sample.
+     */
+    // const datasetId = 'my_dataset'; // Existing dataset
+    // const tableId = 'my_table'; // Existing table
+
+    // Create table reference.
+    const dataset = bigquery.dataset(datasetId);
+    const table = dataset.table(tableId);
+
+    function manualPaginationCallback(err, rows, nextQuery) {
+      console.log('Rows:');
+      rows.forEach(row => console.log(row));
```

Comment:
It'd be helpful to show what operations are possible on rows. Are they indexable by number or column name, for example.

(We don't actually do that in Python https://github.com/googleapis/python-bigquery/blob/9d43d2073dc88140ae69e6778551d140430e410d/samples/browse_table_data.py but if I did it again, I'd prefer listing rows from a sample table with a well-known schema.

Reply:
I think rewriting this sample to be able to do that will introduce complexity that decreases its overall quality. This library ultimately infers the table's project ID from the instance of the client, so creating a reference to a public datasets table in order to call getRows() would require showing users to manually reset the projectId, which would likely create a lot of confusion. Alternately, creating a table using a query would require calling job.getQueryResults() in order to determine when the job is finished creating the table, and would then be confusing because that method also returns the rows.

Reply:
Oh, meant to also add that row values are accessible in the same way that would be expected with objects in Node.js, so I think we'd be safe in opting to not explicitly show it here.

Reply:
Hmmm... I don't like that it's not easier to download rows from across projects, but that shouldn't block this sample.

> job.getQueryResults() in order to determine when the job is finished creating the table, and would then be confusing because that method also returns the rows.

See my other comment. I have a concern if `job.getQueryResults()` downloads all rows. The main point of pagination is to save on memory resources. It's bad if it's not possible to do so with this client for query results. (Maybe there's a feature request hidden in here for a getQueryResultsStream method?)

Reply:
You can paginate. I was just thinking that the reason for using `job.getQueryResults()` in this way might not be clear to users, but I think you're right that it's better to do it that way. Updated.